sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/BufferSendState.scala

```diff
@@ -171,32 +171,8 @@ class BufferSendState(
 
         acquiredBuffs.foreach { case RangeBuffer(blockRange, rapidsBuffer) =>
           require(blockRange.rangeSize() <= bounceBuffToUse.getLength - buffOffset)
-          withResource(rapidsBuffer.getMemoryBuffer) { memBuff =>
-            bounceBuffToUse match {
-              case _: HostMemoryBuffer =>
-                //TODO: HostMemoryBuffer needs the same functionality that
-                // DeviceMemoryBuffer has to copy from/to device/host buffers
-                CudaUtil.copy(
-                  memBuff,
-                  blockRange.rangeStart,
-                  bounceBuffToUse,
-                  buffOffset,
-                  blockRange.rangeSize())
-              case d: DeviceMemoryBuffer =>
-                memBuff match {
-                  case mh: HostMemoryBuffer =>
-                    // host original => device bounce
-                    d.copyFromHostBufferAsync(buffOffset, mh, blockRange.rangeStart,
-                      blockRange.rangeSize(), serverStream)
-                  case md: DeviceMemoryBuffer =>
-                    // device original => device bounce
-                    d.copyFromDeviceBufferAsync(buffOffset, md, blockRange.rangeStart,
-                      blockRange.rangeSize(), serverStream)
-                  case _ => throw new IllegalStateException("What buffer is this")
-                }
-              case _ => throw new IllegalStateException("What buffer is this")
-            }
-          }
+          rapidsBuffer.copyToMemoryBuffer(
```

Comment:
In the GDS scenario, we never spill to host memory, correct?  

Mostly just checking, because the bounce buffer picked here can be host in the non-GDS case, and that would trigger the exception for the GDS RapidsBuffer.